src/parachain/amm.ts

```diff
@@ -594,22 +594,38 @@ export class DefaultAMMAPI implements AMMAPI {
         return { extrinsic: swapExtrinsic, event: this.api.events.dexStable.CurrencyExchange };
     }
 
+    private async _getFarmingRewardCurrencyIds(
+        lpTokenCurrencyId: InterbtcPrimitivesCurrencyId
+    ): Promise<Array<InterbtcPrimitivesCurrencyId>> {
+        const rewardCurrencies: Array<InterbtcPrimitivesCurrencyId> = [];
+
+        const rewardCurrenciesRaw = await this.api.query.farmingRewards.rewardCurrencies(lpTokenCurrencyId);
+        rewardCurrenciesRaw.forEach((rewardCurrencyId: InterbtcPrimitivesCurrencyId) =>
+            rewardCurrencies.push(rewardCurrencyId)
+        );
```

Comment:
Suggestion (not blocking): Maybe something like this will work to turn the btreeset into an array directly?

```
const rewardCurrencies = Array.from(rewardCurrenciesRaw.values());
```